Helper/Data.php

```diff
@@ -1196,4 +1193,62 @@ public function initializeAdyenClient($storeId = null)
 
 		return $client;
 	}
+
+	/**
+	 * Retrieve origin keys for platform's base url
+	 *
+	 * @return string
+	 * @throws \Adyen\AdyenException
+	 */
+	public function getOriginKeyForBaseUrl()
+	{
+		$baseUrl = $this->storeManager->getStore()->getBaseUrl(\Magento\Framework\UrlInterface::URL_TYPE_WEB);
+		$parsed = parse_url($baseUrl);
+		$domain = $parsed['scheme'] . "://" . $parsed['host'];
+
+		if (!$originKey = $this->cache->load("Adyen_origin_key_for_" . $domain)) {
+			$originKey = "";
+
+			if ($originKeys = $this->getOriginKeys($domain)) {
+				$originKey = $originKeys[$domain];
+				$this->cache->save($originKey, "Adyen_origin_key_for_" . $domain, array(), 60 * 60 * 24);
+			}
+		}
+
+		return $originKey;
+	}
+
+	/**
+	 * Get origin keys for a specific url using the adyen api library client
+	 *
+	 * @param $url
+	 * @return mixed
+	 * @throws \Adyen\AdyenException
+	 */
+	private function getOriginKeys($url)
+	{
+		$params = array(
+			"originDomains" => array(
+				$url
+			)
+		);
+
+		$client = $this->initializeAdyenClient();
+
+		$service = $this->adyenServiceFactory->createCheckoutUtility($client);
+		$respone = $service->originKeys($params);
+		return $respone['originKeys'];
```

Comment:
For a given url there should be one originKey, so we better return the orginKey for this domain instead of an array.
This will simplify the logic in L1212
```suggestion
		return current($respone['originKeys']);
```